src/shared/elf-util.c

```diff
@@ -388,6 +388,7 @@ static int parse_buildid(Dwfl_Module *mod, Elf *elf, const char *name, StackCont
         int r;
 
         assert(mod || elf);
+        assert(name);
```

Comment:
same here, if this comes from the other callback, it might not be set (so changing the use in the log messages to `strna` should be the right fix)

Reply:
`parse_buildid()` is called from `parse_elf()` and `module_callback()`, and both sets non-NULL `name` now.

Reply:
It can still be NULL when it comes from parse_core

Reply:
Ah, `package_metadata` is also passed to StackContext... Ouch.

Reply:
You mean parse_core -> module_callback -> parse_buildid ??

Reply:
Yes, because it's called from libdwl, not directly, so we don't know for sure

Reply:
But `module_callback()` sets `name` if it is NULL... (I already dropped the change in `module_callback()`)

Reply:
ah yes, of course